modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -233,6 +256,28 @@ template<> struct vector_type_of<ushort> { using type = v_uint16; };
 template<> struct vector_type_of<short> { using type = v_int16; };
 template<> struct vector_type_of<float> { using type = v_float32; };
 
+template<typename scalar_t>
+struct zero_vec_type_of;
+
+template<typename scalar_t>
+using zero_vec_type_of_t = typename zero_vec_type_of<scalar_t>::type;
+
+template<> struct zero_vec_type_of<uchar> { using type = v_int16; };
```

Comment:
int16 for uchar? is it not mistake?

Reply:
It's not mistake it's implementation specifics.